### PR TITLE
chore(*) improve DP insights API filtering

### PR DESCRIPTION
**Pull request by nickolaev**
### Summary

Improve the API to have better filtering for dataplane insights, such that this request will now work ->
`curl http://localhost:5681/dataplanes+insights?gateway=false&ingress=false`

### CODE REVIEW

**Comment by tharun208** [comment]
@nickolaev, Also we need to add the filter option for `ingress` data planes in the cli.

**Comment by nickolaev** [comment]
> @nickolaev, Also we need to add the filter option for `ingress` data planes in the cli.

Please give an example of the CLI command.

**Comment by tharun208** [comment]
> > @nickolaev, Also we need to add the filter option for `ingress` data planes in the cli.
> 
> Please give an example of the CLI command.

we are already having an option in CLI to filter `gateway` dataplanes - `kumactl inspect dataplanes --gateway`.

**Review by tharun208 — APPROVED** [review]

File: pkg/core/resources/apis/mesh/dataplane_overview.go [hunk 1]
```diff
@@ -125,22 +128,46 @@ func (d *DataplaneOverviewResourceList) RetainMatchingTags(tags map[string]strin
...
+	case "true":
+		return func(a interface{}) bool {
+			return !isnil(a)
+		}
+	case "false":
+		return func(a interface{}) bool {
+			return isnil(a)
+		}
+	default:
+		return func(a interface{}) bool {
```
Comment (lobkovilya) [inline pkg/core/resources/apis/mesh/dataplane_overview.go:145]: If a query has `?gateway=some_garbage_value` does it mean it returns gateways? Can we somehow validate this?

**Review by lobkovilya — APPROVED** [review]

**Pull request merged**